scripts/shared/dbRequestUtils.js

```diff
@@ -0,0 +1,199 @@
+ * @param {Promise} promise - Promise to resolve/reject on a response "end" or
+ *                           "error" event.
+ * @param {String} errorMsg - Optional error message to prepend to the error
+ *                            received from a response "error" event.
+ * @return {ResponseCallback} - Reponse callback function suitable for an http
+ *                              request.
+ */
+gpii.dbRequest.createResponseHandler = function (handleEnd, options, promise, errorMsg) {
+    errorMsg = (errorMsg || "");
+    return function (response) {
+        var responseString = "";
+
+        response.setEncoding("utf8");
+        response.on("data", function (chunk) {
+            responseString += chunk;
+        });
+        response.on("end", function () {
+            if (response.statusCode >= 400) {   // error
+                var fullErrorMsg = errorMsg +
+                                   response.statusCode + " - " +
+                                   response.statusMessage;
+                // Document-not-found or 404 errors include a reason in the
+                // response.
+                // http://docs.couchdb.org/en/stable/api/basics.html#http-status-codes
+                if (response.statusCode === 404) {
+                    fullErrorMsg = fullErrorMsg + ", " +
+                                   JSON.parse(responseString).reason;
+                }
+                promise.reject(fullErrorMsg);
+            }
+            else {
+                var value = handleEnd(responseString, options);
+                promise.resolve(value);
+            }
+        });
+        response.on("error", function (e) {
+            fluid.log(errorMsg + e.message);
+            promise.reject(e);
+        });
+    };
+};
+
+/**
+ * General mechanism to create a database request, set up an error handler and
+ * return.  It is up to the caller to trigger the request by calling its end()
+ * function.
+ * @param {String} databaseURL - URL to query the database with.
+ * @param {ResponseCallback} handleResponse - callback that processes the
+ *                                            response from the request.
+ * @param {String} errorMsg - optional error message for request errors.
+ * @param {Promise} promise - promise to reject on a request error.
+ * @return {http.ClientRequest} - The http request object.
+ */
+gpii.dbRequest.queryDatabase = function (databaseURL, handleResponse, errorMsg, promise) {
+    var aRequest = http.request(databaseURL, handleResponse);
+    aRequest.on("error", function (e) {
+        fluid.log(errorMsg + e.message);
+        promise.reject(e);
+    });
+    return aRequest;
+};
+
+/**
+ * Given a set of database documents, mark each of them for deletion.
+ * @param {Array} docs - Array of records to delete from the database.
+ * @return {Array} - the records marked for deletion.
+ */
+gpii.dbRequest.markDocsForDeletion = function (docs) {
```

Comment:
This utility function can be removed.

Reply:
Yup.

I thought I had, but I missed it.  It's gone now.